Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -64,47 +68,56 @@ public class CustomerIOImplementation: CustomerIOInstance {
     public func identify<RequestBody: Encodable>(
         identifier: String,
         body: RequestBody,
-        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void,
         jsonEncoder: JSONEncoder? = nil
     ) {
-        identifyRepository
-            .addOrUpdateCustomer(identifier: identifier, body: body, jsonEncoder: jsonEncoder) { [weak self] result in
-                DispatchQueue.main.async { [weak self] in
-                    guard self != nil else { return }
-
-                    switch result {
-                    case .success:
-                        return onComplete(Result.success(()))
-                    case .failure(let error):
-                        return onComplete(Result.failure(error))
-                    }
-                }
-            }
+        if let currentlyIdentifiedProfileIdentifier = profileStore.identifier {
+            // TODO: add to background queue delete device token from currently registered profile.
+        }
+
+        let jsonBodyString = jsonAdapter.toJsonString(body, encoder: jsonEncoder)
+
+        let queueTaskData = IdentifyProfileQueueTaskData(identifier: identifier,
+                                                         attributesJsonString: jsonBodyString)
+        let queueStatus = backgroundQueue.addTask(type: .identifyProfile, data: queueTaskData)
+
+        // don't modify the state of the SDK until we confirm we added a background queue task successfully.
+        if queueStatus.success {
+            profileStore.identifier = identifier
+        }
```

Comment:
if this fails to queue, the customer needs to know and be able to act on it immediately.

Reply:
This may fall again under the error handling logic. Let's have a chat about strategy on errors with the SDK and background queue.

Reply:
again, a `TODO` probably helps us catalog these for when that work gets done

Reply:
Done. Also created an issue to come up with a strategy for this https://github.com/customerio/customerio-ios/issues/84